Mathlib/Algebra/Category/ModuleCat/Baer.lean

```diff
@@ -0,0 +1,134 @@
+/-
+Copyright (c) 2025 Nailin Guan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Nailin Guan
+-/
+module
+
+public import Mathlib.Algebra.Category.Grp.Zero
+public import Mathlib.Algebra.Category.ModuleCat.EnoughInjectives
+public import Mathlib.Algebra.Category.ModuleCat.Projective
+public import Mathlib.Algebra.Homology.DerivedCategory.Ext.EnoughInjectives
+public import Mathlib.Algebra.Homology.DerivedCategory.Ext.EnoughProjectives
+public import Mathlib.Algebra.Homology.ShortComplex.ModuleCat
+public import Mathlib.RingTheory.Ideal.Quotient.Operations
+
+/-!
+# Category Language Baer Criterion
+-/
+
+@[expose] public section
+
+universe u u' v v'
+
+variable {R : Type u} [CommRing R]
+
+open CategoryTheory Abelian
+
+private instance small_of_quotient [Small.{v} R] (I : Ideal R) : Small.{v} (R ⧸ I) :=
+  small_of_surjective Ideal.Quotient.mk_surjective
+
+section
+
+universe w
+
+variable [UnivLE.{v, w}]
+
+private instance hasExt_of_small [Small.{v} R] : CategoryTheory.HasExt.{w} (ModuleCat.{v} R) :=
+  CategoryTheory.hasExt_of_enoughProjectives.{w} (ModuleCat.{v} R)
+
+open Limits in
+lemma injective_of_subsingleton_ext_quotient_one [Small.{v} R] (M : ModuleCat.{v} R)
+    (h : ∀ (I : Ideal R), Subsingleton (Ext.{w} (ModuleCat.of R (Shrink.{v, u} (R ⧸ I))) M 1)) :
+    Injective M := by
```

Comment:
It would be nice to make a separate `iff` lemma giving the concrete condition under which `Subsingleton (Ext (ModuleCat.of R (Shrink.{v, u} (R ⧸ I))) M 1))` hold for a single ideal `I`.
Also, with my suggestion to only use the universe `v` for `Ext`, it becomes possible to state that there is an exact sequence `M -> Hom(I, M) -> Ext^1(R/I, M) -> 0` in `ModuleCat.{v} R`.